### PR TITLE
Configure & apply isort via pre-commit

### LINKED ISSUE #350: Sorting import statements
The ordering of `import` statements in the code is all over the place.  Fortunately, there are two major projects that can help out in different ways (using one or the other, not both):

* [`flake8-import-order`](https://github.com/PyCQA/flake8-import-order) — a plugin for flake8 that, if used, would cause the linter to fail if any imports are out of order
    * The plugin supports a variety of import order styles.  I would recommend either the `appnexus` style or [my custom `jwodder` style](https://pypi.org/project/flake8-import-order-jwodder/).
* [`isort`](https://github.com/PyCQA/isort) — a program that can be added to the pre-commit configuration to automatically edit source files to put `import` statements in a reasonable order.
    * I recommend the following configuration for isort:

            profile = "black"
            force_sort_within_sections = true
            known_first_party = ["dandi"]

### PR DESCRIPTION
**Pull request by jwodder**
Closes #350.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -13,3 +13,8 @@ repos:
     hooks:
     -   id: black
         exclude: ^(dandi/_version\.py|versioneer\.py)$
+-   repo: https://github.com/PyCQA/isort
+    rev: 5.7.0
+    hooks:
+    -   id: isort
+        exclude: ^(dandi/_version\.py|versioneer\.py)$
```
Comment (yarikoptic) [inline .pre-commit-config.yaml:20]: also exclude `dandi/due.py` please

**Review by yarikoptic — CHANGES_REQUESTED** [review]
minor request  + should not we also list it in `setup.cfg` for `style`?

**Comment by yarikoptic** [comment]
and I guess we should merge it asap (I assume 3.9 fails are all due to #352 , didn't look)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jwodder) [reply .pre-commit-config.yaml:20]: Do you want that to be excluded from black as well?

**Comment by jwodder** [comment]
> should not we also list it in setup.cfg for style?

pre-commit takes care of installing isort itself, so only pre-commit needs to be listed under `style`.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (yarikoptic) [reply .pre-commit-config.yaml]: ideally - yes, I thought it was so.  It is a "standard" stab file from duecredit, we might take a new version in some time in the future.  I guess I have missed adding it to exclusions when I have added it.

**Comment by yarikoptic** [comment]
all green (but 3.9) - let's proceed to not accumulate conflicts

**Pull request merged**